### DISCUSSION TITLE
How to wait for AddScriptToExecuteOnDocumentCreated completion (C++)

### DISCUSSION THREAD

**Discussion by cylvoon** [MicrosoftEdge/WebView2Feedback#2268] [Q&A]
In a C++ (MFC) app using WebView2, I can't find a way to simply wait until the script passed to AddScriptToExecuteOnDocumentCreated is ready. My ICoreWebView2AddScriptToExecuteOnDocumentCreatedCompletedHandler is just never called if I add some waiting code (e.g. WaitForSingleObject) after calling AddScriptToExecuteOnDocumentCreated.
Without the wait, the ICoreWebView2AddScriptToExecuteOnDocumentCreatedCompletedHandler is reached and I can see it's invoked on the message thread. Then, obviously, if I have to make the UI (i.e. message thread) wait, it can't work.

Does someone know how we're supposed to use this properly?

**Comment by victorthoang** [comment]
hello @cylvoon 

Are you still looking for an answer to your question? If you still need a response, we can turn this into an Issue with a Question tag (which is typically how we address questions and bugs.

The discussion thread might be more useful for off topic questions or non-issue related inquiries!

Let me know if you'd like me to create an issue from this discussion, or if you'd prefer to add it. Thanks!
